controller/controller.go

```diff
@@ -62,8 +68,41 @@ const (
 
 	// DefaultIngressThreads is the default number of ingress worker threads to start with the controller
 	DefaultIngressThreads = 10
+
+	// DefaultLeaderElect is the default true leader election should be enabled
+	DefaultLeaderElect = true
+
+	// DefaultLeaderElectionNamespace is the default namespace used to perform leader election. Only used if leader election is enabled
+	DefaultLeaderElectionNamespace = "kube-system"
+
+	// DefaultLeaderElectionLeaseDuration is the default time in seconds that non-leader candidates will wait to force acquire leadership
+	DefaultLeaderElectionLeaseDuration = 15 * time.Second
```

Comment:
can you provide proposal/discussion on how this works?

Reply:
I added some [description](https://github.com/argoproj/argo-rollouts/pull/1519/files#diff-0bb42a85d04c8b7c9b659307979fd43e8f6e084283a300bfff959adfcbddfd17) to the PR about how the implement work . Please let me know if more detail is needed. Thanks.